src/swtpm_setup/swtpm.c

```diff
@@ -132,26 +130,29 @@ static int swtpm_start(struct swtpm *self)
     if (!success) {
         logerr(self->logfile, "Could not start swtpm: %s\n", error->message);
         g_error_free(error);
-        return 1;
+        goto error;
     }
 
-    /* wait until the pidfile appears or swtpm terminates */
+    /* wait until the pidfile is written to or swtpm terminates */
     for (ctr = 0; ctr < 1000; ctr++) {
         if (kill(self->pid, 0) < 0) {
             /* swtpm terminated */
             self->pid = 0;
             logerr(self->logfile, "swtpm process terminated unexpectedly.\n");
             self->cops->stop(self);
-            return 1;
+            goto error;
         }
-        if (stat(self->pidfile, &statbuf) == 0) {
+        if (stat(self->pidfile, &statbuf) == 0 && statbuf.st_size > 0) {
             printf("TPM is listening on Unix socket.\n");
-            return 0;
+            ret = 0;
+            break;
         }
         usleep(5000);
     }
 
-    return 1;
+error:
+    close(pidfile_fd);
```

Comment:
This function should remove the pidfile if swtpm isn't running once it returns.

Reply:
it might actually be smarter to just do away with `swtpm_remove_pidfile` and unlink it immediately, that way no leftovers are possible - afaict this isn't used anywhere else, and I don't think there's much user value in keeping a pidfile for such a short-lived instance visible